### PR TITLE
IndustryStandard\IpmiNetFnSensorEvent.h: Added SetSensorThreshold and GetSensorThreshold commands

### PR DESCRIPTION
**Pull request by apop5**
## Description

[The Ipmi Specification](https://www.intel.com/content/www/us/en/products/docs/servers/ipmi/ipmi-second-gen-interface-spec-v2-rev1-1.html) contains a lot of commands and structures that are not widely used in firmware. 

PR to add the Set Sensor Thresholds and Get Sensor Thresholds commands, and their associated structure definitions.

<_Please include a description of the change and why this change was made._>

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested
These are just command and structure definitions. 

## Integration Instructions
N/A

### CODE REVIEW

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h [hunk 1]
```diff
@@ -42,5 +42,37 @@ typedef struct {
...
 } IPMI_PLATFORM_EVENT_MESSAGE_DATA_REQUEST;
 
+//
+// Definitions for Set Sensor Thresholds command
+//
+#define IPMI_SENSOR_SET_SENSOR_THRESHOLDS  0x26
+
+typedef struct _IPMI_SENSOR_SET_SENSOR_THRESHOLD_REQUEST_DATA {
+  UINT8    SensorNumber;
+  UINT8    SetBitEnable;
```
Comment (cfernald) [inline MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]: Should be add #defines for the set bit enable options?

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h [hunk 2]
```diff
@@ -42,5 +42,37 @@ typedef struct {
...
+} IPMI_SENSOR_SET_SENSOR_THRESHOLD_REQUEST_DATA;
+
+//
+// Definitions for Get Sensor Thresholds command
+//
+#define IPMI_SENSOR_GET_SENSOR_THRESHOLDS  0x27
+
+typedef struct _IPMI_SENSOR_GET_SENSOR_THRESHOLD_RESPONSE_DATA {
+  UINT8    CompletionCode;
+  UINT8    SetBitEnable;
```
Comment (cfernald) [inline MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]: Should this be GetBitEnable in this case since its the read not the write?

**Review by cfernald — APPROVED** [review]

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h (hunk 1 above)
Comment (apop5) [reply MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]: Would you prefer defines, or should I follow IpmiNetFnChassis.h's IPMI_POWER_RESTORE_POLICY and do bitwise definitions?

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h (hunk 2 above)
Comment (apop5) [reply MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]: Yes. Depending on the response to https://github.com/microsoft/mu_basecore/pull/288/files#r1120671605, I will either change it to GetBitEnable, or I will change it to the resulting union.

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h (hunk 1 above)
Comment (cfernald) [reply MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]: I like the bitfield approach you did

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h [hunk 3]
```diff
@@ -42,5 +42,50 @@ typedef struct {
...
+  } Bits;
+  UINT8    Uint8;
+} SENSOR_BITS;
+
+typedef struct _IPMI_SENSOR_SET_SENSOR_THRESHOLD_REQUEST_DATA {
+  UINT8          SensorNumber;
+  SENSOR_BITS    SetBitEnable;
+  UINT8          LowerNonCriticalThreshold;
+  UINT8          LowerCriticalThreshold;
+  UINT8          LowerNonRecoverablethreshold;
```
Comment (makubacki) [inline MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]:
```suggestion
  UINT8          LowerNonRecoverableThreshold;

```

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h [hunk 4]
```diff
@@ -42,5 +42,50 @@ typedef struct {
...
+//
+// Definitions for Set Sensor Thresholds command
+//
+#define IPMI_SENSOR_SET_SENSOR_THRESHOLDS  0x26
+
+typedef union {
+  struct _SENSOR_BITS {
+    UINT8    LowerNonCriticalThreshold    : 1;
+    UINT8    LowerCriticalThreshold       : 1;
+    UINT8    LowerNonRecoverablethreshold : 1;
```
Comment (makubacki) [inline MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]:
```suggestion
    UINT8    LowerNonRecoverableThreshold : 1;

```

File: MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h [hunk 5]
```diff
@@ -42,5 +42,50 @@ typedef struct {
...
+
+typedef struct _IPMI_SENSOR_GET_SENSOR_THRESHOLD_RESPONSE_DATA {
+  UINT8          CompletionCode;
+  SENSOR_BITS    GetBitEnable;
+  UINT8          LowerNonCriticalThreshold;
+  UINT8          LowerCriticalThreshold;
+  UINT8          LowerNonRecoverablethreshold;
+  UINT8          UpperNonCriticalThreshold;
+  UINT8          UpperCriticalThreshold;
+  UINT8          UpperNonRecoverablethreshold;
```
Comment (makubacki) [inline MdePkg/Include/IndustryStandard/IpmiNetFnSensorEvent.h]:
```suggestion
  UINT8          LowerNonRecoverableThreshold;
  UINT8          UpperNonCriticalThreshold;
  UINT8          UpperCriticalThreshold;
  UINT8          UpperNonRecoverableThreshold;

```

**Review by makubacki — APPROVED** [review]

**Pull request merged**